root.go

```diff
@@ -26,18 +43,104 @@ func (r Root) AddPcap(path string, limit int, warner zbuf.Warner) (nano.Span, er
 		return nano.Span{}, err
 	}
 
+	return index.Span(), r.AddPcapWithIndex(path, index)
+}
+
+func (r Root) AddPcapWithIndex(path string, index pcap.Index) (err error) {
+	path, err = filepath.Abs(path)
+	if err != nil {
+		return err
+	}
 	// symlink path to root
 	if err := os.Symlink(path, r.SymlinkPath(path)); err != nil {
-		return nano.Span{}, err
+		return err
 	}
 
 	// create pcap index
-	if err = fs.MarshalJSONFile(index, r.IndexPath(path), 0644); err != nil {
+	if err := fse.MarshalJSONFile(index, r.IndexPath(path), 0644); err != nil {
 		r.DeletePcap(path)
-		return nano.Span{}, err
+		return err
+	}
+	return nil
+}
+
+func (r Root) Search(ctx context.Context, req Search, w io.Writer) error {
+	var search pcap.Search
+	// We add two microseconds to the end of the span as fudge to deal with the
+	// fact that zeek truncates timestamps to microseconds where pcap-ng
+	// timestamps have nanosecond precision.  We need two microseconds because
+	// both the base timestamp of a conn record as well as the duration time
+	// can be truncated downward.
+	span := nano.NewSpanTs(req.Span.Ts, req.Span.End()+2000)
+	flow := pcap.NewFlow(req.SrcIP, int(req.SrcPort), req.DstIP, int(req.DstPort))
+	switch req.Proto {
+	case "tcp":
+		search = pcap.NewTCPSearch(span, flow)
+	case "udp":
+		search = pcap.NewUDPSearch(span, flow)
+	case "icmp":
+		search = pcap.NewICMPSearch(span, req.SrcIP, req.DstIP)
+	default:
+		return fmt.Errorf("unsupported proto type: %s", req.Proto)
 	}
 
-	return index.Span(), nil
+	files, err := r.Pcaps()
+	if err != nil {
+		return err
+	}
+
+	group, _ := errgroup.WithContext(ctx)
```

Comment:
```suggestion
	group, cancel := errgroup.WithContext(ctx)
	defer concel()
```

Reply:
This returns a context.Context not a context.CancelFunc